dra-evolution/pkg/api/claim_types.go

```diff
@@ -361,12 +402,28 @@ type RequestAllocationResult struct {
 	// device to be allocated.
 	RequestName string `json:"requestName"`
 
-	// DeviceName references one device instance via its name in the resource
-	// pool. Driver name and node name have to be determined from the
-	// context.
+	// This name together with the driver name and the device name field
+	// identify which device was allocated.
+	//
+	// For a driver with node-local devices, it is the name of the node and
+	// the overall ID is `<driver name>/<node name>/<device name>`.
+	//
+	// For other drivers, the overall ID is `<driver name>/<device pool
+	// name>/<device name>`.
+	//
+	// Must not be longer than 253 characters and may contain one or more
+	// DNS sub-domains separated by slashes.
+	//
+	// +optional
+	DevicePoolName string `json:"devicePoolName,omitempty"`
```

Comment:
When I see this I think of out whole `ResourcePool` and `DevicePool` discussion and it makes me think this is referring to the name of the `ResourcePool` that the device in this result was allocated from. I think it is probably meant to be something different though, but the previous discussions are carrying too much baggage for me to understand exactly what that is.

Reply:
https://github.com/kubernetes-sigs/wg-device-management/pull/26#discussion_r1632105483